feat(curriculum): add cipher project to Python curriculum (#52373)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/6554a49a4f782f208abcc87e.md

---
id: 6554a49a4f782f208abcc87e
title: Step 58
challengeType: 20
dashedName: step-58
---

# --description--

> You need to increase the `key_index` count, so that it is updated for the next iteration.

Reviewer: ```suggestion
You will need to increase the `key_index` count for the next iteration.
```

> After the line you've just added, and in the same block, use the addition assignment operator to increment `key_index` by one.

Reviewer: ```suggestion
To do this, after the line you just added and in the same code block, use the addition assignment operator to increment `key_index` by one.
```

# --hints--

You should use the `+=` operator to add `1` to `key_index`.

```js
({ test: () => assert.match(code, /key_index\s*\+=\s*1/) })
```

# --seed--

## --seed-contents--

```py

text = 'Hello Zaira'
custom_key = 'python'

def vigenere(message, key):
    key_index = 0
    alphabet = 'abcdefghijklmnopqrstuvwxyz'
    encrypted_text = ''

    for char in message.lower():
--fcc-editable-region--
        # Append space to the message
        if char == ' ':
            encrypted_text += char

> 
>         else:

Reviewer: I would suggest removing this blank line before `else`:
```suggestion
        else:
```
